syncanocli/context.py

```diff
@@ -0,0 +1,138 @@
+from ConfigParser import SafeConfigParser
+
+import os
+import click
+import six
+
+from . import settings
+
+
+class Echo(object):
+    '''Object oriented wrapper around Click `secho` function.'''
+
+    def __init__(self, verbose=0):
+        self.verbose = verbose
+
+    def __call__(self, message, verbose=None, **style):
+        if self.verbose <= 0:
+            return
+
+        if verbose and self.verbose < verbose:
+            return
+
+        click.secho(message, **style)
+
+    def error(self, *args, **kwargs):
+        kwargs['bold'] = True
+        kwargs['fg'] = 'red'
+        self(*args, **kwargs)
+
+    def warning(self, *args, **kwargs):
+        kwargs['bold'] = True
+        kwargs['fg'] = 'yellow'
+        self(*args, **kwargs)
+
+    def info(self, *args, **kwargs):
+        kwargs['bold'] = False
+        kwargs['fg'] = 'blue'
+        self(*args, **kwargs)
+
+    def success(self, *args, **kwargs):
+        kwargs['bold'] = False
+        kwargs['fg'] = 'green'
+        self(*args, **kwargs)
+
+
+class ConfigSection(object):
+
+    def __init__(self, config, section):
+        self._config = config
+        self._section = section
+
+    def __getattr__(self, k):
+        if k.startswith('_'):
+            return self.__getattribute__(k)
+        return self._config._data[self._section].get(k)
+
+    def __setattr__(self, k, v):
+        if k.startswith('_'):
+            object.__setattr__(self, k, v)
+        else:
+            self._config._data[self._section][k] = v
+
+    def __delattr__(self, k):
+        if k in self._config._data[self._section]:
+            del self._config._data[self._section][k]
+
+
+class Config(object):
+
+    def __init__(self, filename):
+        self.filename = filename
+        self._data = {}
+        self.read()
+
+    def __getattr__(self, k):
+        if k not in self.__dict__:
+            if k not in self._data:
+                self._data[k] = {}
+
+            section = ConfigSection(self, k)
```

Comment:
syntax sugar u can access all config properties via objects:(dot notation) e.g: `config.credentials.api_key` this is corresponding to ini files structure e.g `config.{section}.{variable}`